### PR TITLE
Backups

### PR DESCRIPTION
**Pull request by ferdinandpolpol**

### CODE REVIEW

File: docker-compose.override.yml [hunk 1]
```diff
@@ -7,6 +7,18 @@ services:
...
+      - ./conf/nginx/conf.d/dev.conf:/etc/nginx/conf.d/oscar.template
+
+  backup:
+    build: .
+    image: countable/s3-backup-job:1.0
+    volumes:
+      - $HOME/.aws/credentials:/root/.aws/credentials
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      - BACKUP_BUCKET=countable/backups
```
Comment (countable) [inline docker-compose.override.yml]: This bucket would need to be an ENV var, not defaulting to our account

File: docker-compose.override.yml [hunk 2]
```diff
@@ -7,6 +7,18 @@ services:
       - '443:443'
     volumes:
       - ./conf/ssl:/ssl
+      - ./conf/nginx/conf.d/dev.conf:/etc/nginx/conf.d/oscar.template
+
+  backup:
+    build: .
+    image: countable/s3-backup-job:1.0
+    volumes:
+      - $HOME/.aws/credentials:/root/.aws/credentials
```
Comment (countable) [inline docker-compose.override.yml]: Can you document the step for setting up credentials and everything else for backups, in the README?

File: docker-compose.override.yml [hunk 3]
```diff
@@ -7,6 +7,18 @@ services:
...
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      - BACKUP_BUCKET=countable/backups
+      - BACKUP_CMD='mysqldump -uroot -pliyi oscar_mcmaster > dump'
+    restart: always
```
Comment (countable) [inline docker-compose.override.yml]: We will also need to move the OscarDocuments folder to the backup, and set the frequency to daily instead of hourly.

**Comment by ferdinandpolpol** [comment]
@countable changed the image to a custom backup job forked from s3-backup-jobs `https://github.com/ferdinandpolpol/s3-backup-jobs/commit/1c4060c9989e2f5a181a3d360c02c1ccfd1e7459`, docker hub `https://hub.docker.com/repository/docker/ferdinandpolpol/s3-oscar-backup/`. Is this alright?

File: docker-compose.override.yml (hunk 2 above)
Comment (ferdinandpolpol) [reply docker-compose.override.yml]: added

File: docker-compose.override.yml (hunk 1 above)
Comment (countable) [reply docker-compose.override.yml]: @ferdinandpolpol can you remove Countable from this path?

File: docker-compose.override.yml (hunk 3 above)
Comment (countable) [reply docker-compose.override.yml]: @ferdinandpolpol can we include a bash script to do this?

Comment (ferdinandpolpol) [reply docker-compose.override.yml]: Do you mean a script to manually execute the backups?

Comment (countable) [reply docker-compose.override.yml]: Instead of mysqldump, we need a script to do that and also zip up and copy the OscarDocuments folder to s3.

Comment (ferdinandpolpol) [reply docker-compose.override.yml]: @countable I added the OscarDocuments backup here https://github.com/ferdinandpolpol/s3-backup-jobs/commit/1c4060c9989e2f5a181a3d360c02c1ccfd1e7459 , we are using my fork for the backup image.

**Pull request merged**